components/ui/pathogen-map/use-pathogen-map-mouse-events.ts

```diff
@@ -0,0 +1,70 @@
+import { useState } from "react";
+
+import { PathogenDataPointPropertiesBase } from "./pathogen-map";
+import { PopupInfo } from "./pathogen-map-popup";
+import { PathogenMapLayerInfo } from "./pathogen-map-layer";
+
+export type PathogenMapCursor = "" | "pointer";
+
+interface UsePathogenMapMouseProps<TPathogenDataPointProperties extends PathogenDataPointPropertiesBase> {
+  baseCursor: PathogenMapCursor;
+  layers: PathogenMapLayerInfo<TPathogenDataPointProperties>[];
+  setPopUpInfo: (input: PopupInfo<TPathogenDataPointProperties>) => void
+}
+
+export const usePathogenMapMouse = <TPathogenDataPointProperties extends PathogenDataPointPropertiesBase>({ baseCursor, layers, setPopUpInfo }: UsePathogenMapMouseProps<TPathogenDataPointProperties>) => {
+  const [cursor, setCursor] = useState<string>(baseCursor);
```

Comment:
For the record, all this `cursor`, `setCursor`, `onMouseEnter`, `onMouseLeave` stuff is for is so that the cursor has the right icon when you hover over a point on the map.

Docs: https://visgl.github.io/react-map-gl/docs/api-reference/map#onmouseenter